lightning: add add-index-by-sql (#12948)
pingcap/docs: tidb-lightning/tidb-lightning-physical-import-mode.md

2. TiDB Lightning creates table schemas in the target database and fetches the metadata.

>     Starting from v7.0.0, `tidb-lightning` adds indexes using the SQL interface by default, and removes all secondary indexes from the target table before importing the data.

Reviewer: ```suggestion
    Starting from v7.0.0, `tidb-lightning` adds indexes calling the SQL interface by default, and drops all secondary indexes from the target table before importing the data.
```

Reviewer 2: I'm not sure which expression is more accurate in English. @hfxsd PTAL

Reviewer 2: In addition, `Starting from v7.0.0` -> `Starting from TiDB v7.0.0`.

Author: > I'm not sure which expression is more accurate in English. @hfxsd PTAL

就用 Frank 这个吧

>     Starting from TiDB v7.0.0, `tidb-lightning` adds indexes calling the SQL interface by default, and drops all secondary indexes from the target table before importing the data.

Reviewer 3: ```suggestion
    Starting from TiDB v7.0.0, `tidb-lightning` adds indexes via the SQL interface by default, and drops all secondary indexes from the target table before importing the data.
```

Reviewer 3: via 或者 by calling

    The engine file contains two types of engines: **data engine** and **index engine**. Each engine corresponds to a type of key-value pairs: row data and secondary index. Normally, row data is completely ordered in the data source, and the secondary index is unordered. Therefore, the data engine files are imported immediately after the corresponding block is written, and all index engine files are imported only after the entire table is encoded.

>     Note that when `tidb-lightning` adds indexes using the SQL interface, the index engine will not write data because the secondary indexes of the target table have already been removed in step 2.

Reviewer: ```suggestion
    Note that when `tidb-lightning` adds indexes calling the SQL interface, the index engine will not write data because the secondary indexes of the target table have already been dropped in step 2.
```

>     Note that when `tidb-lightning` adds indexes calling the SQL interface, the index engine will not write data because the secondary indexes of the target table have already been dropped in step 2.

Reviewer 3: ```suggestion
    Note that when `tidb-lightning` adds indexes via the SQL interface, the index engine will not write data because the secondary indexes of the target table have already been dropped in step 2.
```

> 6. After all engine files are imported, TiDB Lightning compares the checksum between the local data source and the downstream cluster, and ensures that the imported data is not corrupted. Then TiDB Lightning adds the missing secondary indexes, or lets TiDB analyze the new data (`ANALYZE`) to optimize the future operations. Meanwhile, `tidb-lightning` adjusts the `AUTO_INCREMENT` value to prevent conflicts in the future.

Reviewer: ```suggestion
6. After all engine files are imported, TiDB Lightning compares the checksum between the local data source and the downstream cluster, and ensures that the imported data is not corrupted. Then TiDB Lightning adds the previously dropped secondary indexes, or lets TiDB analyze the new data (`ANALYZE`) to optimize the future operations. Meanwhile, `tidb-lightning` adjusts the `AUTO_INCREMENT` value to prevent conflicts in the future.
```

Reviewer: @gozssky PTAL

> 6. After all engine files are imported, TiDB Lightning compares the checksum between the local data source and the downstream cluster, and ensures that the imported data is not corrupted. Then TiDB Lightning adds the previously dropped secondary indexes, or lets TiDB analyze the new data (`ANALYZE`) to optimize the future operations. Meanwhile, `tidb-lightning` adjusts the `AUTO_INCREMENT` value to prevent conflicts in the future.

Author: ```suggestion
6. After all engine files are imported, TiDB Lightning compares the checksum between the local data source and the downstream cluster, and ensures that the imported data is not corrupted. Then TiDB Lightning adds the previously dropped secondary indexes in Step 2, or lets TiDB analyze the new data (`ANALYZE`) to optimize the future operations. Meanwhile, `tidb-lightning` adjusts the `AUTO_INCREMENT` value to prevent conflicts in the future.
```

> 6. After all engine files are imported, TiDB Lightning compares the checksum between the local data source and the downstream cluster, and ensures that the imported data is not corrupted. Then TiDB Lightning adds the previously dropped secondary indexes in Step 2, or lets TiDB analyze the new data (`ANALYZE`) to optimize the future operations. Meanwhile, `tidb-lightning` adjusts the `AUTO_INCREMENT` value to prevent conflicts in the future.

Reviewer 3: ```suggestion
6. After all engine files are imported, TiDB Lightning compares the checksum between the local data source and the downstream cluster, and ensures that the imported data is not corrupted. Then TiDB Lightning adds the previously dropped secondary indexes in step 2, or lets TiDB analyze the new data (`ANALYZE`) to optimize future operations. Meanwhile, `tidb-lightning` adjusts the `AUTO_INCREMENT` value to prevent conflicts in the future.
```
